helloswift/main.swift

```diff
@@ -108,6 +149,16 @@ func checkIfTheCatIsAlive(cat: Cat) {
  - Note:  at this point, the cat should be either alive or dead; but not both
  */
 
+let cat: Cat = Cat(name: "Jellie", isAlive: nil)
+let tnt: TNT = TNT()
+let underground = Underground(occupent: cat, explosive: tnt)
+
+print(underground.occupent.isAlive)
+
+underground.open()
+print(underground.occupent.isAlive!) // should we require the ! as it should not be nil?
```

Comment:
Yup, this variable needs to be forced unwrapped, but its best to handle this with a exception.

```swift
val isCatAlive =  underground.occupent.isAlive ?? throw Error("Cat has to be either alive or dead")
```